src/cplscheme/SerialCouplingScheme.cpp

```diff
@@ -77,13 +77,14 @@ void SerialCouplingScheme::receiveAndSetTimeWindowSize()
 
 void SerialCouplingScheme::performReceiveOfFirstAdvance()
 {
-  if (doesFirstStep()) {
-    // do nothing
-  } else { // second participant
+  if (!doesFirstStep()) { // second participant
+    // similar to SerialCouplingScheme::exchangeSecondData()
     receiveAndSetTimeWindowSize();
     PRECICE_DEBUG("Receiving data...");
     receiveData(getM2N(), getReceiveData());
-    checkDataHasBeenReceived();
+    if (not hasDataBeenReceived()) {
+      checkDataHasBeenReceived();
+    }
```

Comment:
If we already know that data has been received, why do we have to check it?

Reply:
Because for the second participant in a `SerialCouplingScheme` there are two places where data can be received: When we initialize or when we receive it from the other participant's advance. We have to check whether `hasDataBeenReceived()` is true before calling `checkDataHasBeenReceived()`, because an assertion is triggered, if `checkDataHasBeenReceived` is called twice (safeguard).

Reply:
ok, thanks, makes more sense now. please add this information as a comment here

Reply:
Done. 9a518593e72e6e600c2cf5397ece974ecd3b0aaa